editor/components/media-placeholder/index.js

```diff
@@ -63,13 +64,18 @@ class MediaPlaceholder extends Component {
 	}
 
 	onFilesUpload( files ) {
-		const { onSelect, type, multiple, onError } = this.props;
+		const { onSelect, type, multiple, onError = noop, disableMaxUploadErrorMessages = false, noticeOperations } = this.props;
 		const setMedia = multiple ? onSelect : ( [ media ] ) => onSelect( media );
 		editorMediaUpload( {
 			allowedType: type,
 			filesList: files,
 			onFileChange: setMedia,
-			onError,
+			onError: ( errorMessage ) => {
+				onError( errorMessage );
+				if ( ! disableMaxUploadErrorMessages ) {
```

Comment:
I'd be tempted to add a one-line comment here explaining why we check for a false "disabled" prop as without seeing it being used by the component it might be a bit strange to read. Maybe you could link to my comment in the PR or just say:

```js
// We use a prop named `disable`, set to `false` by default, because it makes for a nicer
// component prop API. eg:
//
// <SomeComponent disableMaxUploadErrorMessages />
// 
// instead of:
// 
// <SomeComponent enableMaxUploadErrorMessages={ false } />
```

Just a thought.

Reply:
Also I'd find this check a bit more readable if it were `disableMaxUploadErrorMessages === false` because:

1. I think it communicates the intent a bit more clearly
2. It's a boolean anyway and the default is `false` so it's fine